docs/src/main/asciidoc/opentelemetry.adoc

```diff
@@ -210,6 +210,103 @@ Then visit the http://localhost:16686[Jaeger UI] to see the tracing information.
 
 Hit `CTRL+C` to stop the application.
 
+== Additional configuration
+Some use cases will require custom configuration of OpenTelemetry.
+These sections will outline what is necessary to properly configure it.
+
+=== ID Generator
+The OpenTelemetry extension will use by default a random https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#id-generators[ID Generator]
+when creating the trace and span identifier.
+
+Some vendor-specific protocols needs a custom ID Generator, you can override the default one by creating a producer.
+The OpenTelemetry extension will detect the `IdGenerator` CDI bean and will use it when configuring the tracer producer.
+
+[source,java]
+----
+@Singleton
+public class CustomConfiguration {
+
+    /** Creates a custom IdGenerator for OpenTelemetry */
+    @Produces
+    @Singleton
+    public IdGenerator idGenerator() {
+        return AwsXrayIdGenerator.getInstance();
+    }
+}
+----
+
+=== Propagators
+OpenTelemetry propagates cross-cutting concerns through https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/api-propagators.md[propagators] that will share an underlying `Context` for storing state and accesing
+data across the lifespan of a distributed transaction.
+
+By default, the OpenTelemetry extension enables the https://www.w3.org/TR/trace-context/[W3C Trace Context] and the https://www.w3.org/TR/baggage/[W3C Baggage]
+propagators, you can however choose any of the supported OpenTelemetry propagators by setting the `propagators` config that is described in the <<configuration-reference>>.
+
+[NOTE]
+====
```

Comment:
Should we explicitly provide examples of those dependencies to make it easier to add to a `pom.xml`?

Reply:
Yup =] added